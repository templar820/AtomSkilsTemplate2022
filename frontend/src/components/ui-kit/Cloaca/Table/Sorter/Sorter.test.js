import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import serializer from 'jest-emotion';
import { Table, Thead, Tbody, Row, Cell, Sorter } from '../';

Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(serializer);

const mockFn = jest.fn();

const props = {
    sequence: ['normal', 'asc', 'desc'],
    controlled: false,
    disabled: false,
    onClick: mockFn
};

const mountComponent = (props) =>
    mount(
        <Table>
            <Thead>
                <Row>
                    <Cell leftSlot={<Sorter {...props} />}>H1</Cell>
                </Row>
            </Thead>
            <Tbody>
                <Row>
                    <Cell>1</Cell>
                </Row>
                <Row>
                    <Cell>2</Cell>
                </Row>
            </Tbody>
        </Table>
    );

describe('<Sorter />', () => {
    let table, sorter;

    beforeEach(() => {
        table = mountComponent(props);

        sorter = table.find(Sorter);
    });

    afterEach(() => {
        table.unmount();
    });

    test('should match snapshot', () => {
        expect(toJson(sorter)).toMatchSnapshot();
    });

    test('should render correctly', () => {
        expect(sorter.prop('sequence')).toEqual(props.sequence);
        expect(sorter.prop('controlled')).toEqual(props.controlled);
        expect(sorter.prop('disabled')).toEqual(props.disabled);
    });

    test('should call onClick method on disabled', () => {
        table = mountComponent({ ...props, disabled: true });
        sorter = table.find(Sorter);

        sorter.simulate('click');
        expect(mockFn).not.toHaveBeenCalled();
    });

    test('should call onClick method', () => {
        sorter.simulate('click');
        expect(mockFn).toHaveBeenCalled();
    });

    test('should not sort', () => {
        table = mountComponent({ ...props, disabled: false, onClick: null, controlled: true });
        sorter = table.find(Sorter);

        const firstValue = table
            .find('ChildrenWrapper')
            .at(0)
            .text();

        sorter.simulate('click');

        expect(
            table
                .find('ChildrenWrapper')
                .at(0)
                .text()
        ).toEqual(firstValue);
    });
});
