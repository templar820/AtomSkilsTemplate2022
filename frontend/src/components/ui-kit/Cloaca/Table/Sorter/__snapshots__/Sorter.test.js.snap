// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sorter /> should match snapshot 1`] = `
.emotion-0 {
  cursor: pointer;
}

<Sorter
  controlled={false}
  disabled={false}
  onClick={[MockFunction]}
  sequence={
    Array [
      "normal",
      "asc",
      "desc",
    ]
  }
>
  <SorterSVG
    controlled={false}
    currentDirection="normal"
    disabled={false}
    onClick={[Function]}
  >
    <StyledWrapper
      as="svg"
      controlled={false}
      currentDirection="normal"
      disabled={false}
      fill="none"
      height="20"
      onClick={[Function]}
      viewBox="0 0 20 20"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <svg
        className="emotion-0 emotion-1"
        disabled={false}
        fill="none"
        height="20"
        onClick={[Function]}
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 6L12.5981 9.75H7.40192L10 6Z"
          fill="#D8D8D8"
        />
        <path
          d="M10 15L7.40192 11.25L12.5981 11.25L10 15Z"
          fill="#D8D8D8"
        />
      </svg>
    </StyledWrapper>
  </SorterSVG>
</Sorter>
`;
